salt/modules/zypper.py

```diff
@@ -2033,3 +2123,85 @@ def list_installed_patches():
         salt '*' pkg.list_installed_patches
     '''
     return _get_patches(installed_only=True)
+
+def list_provides(**kwargs):
+    '''
+    List package provides currently installed as a dict.
+    {'<provided_name>': ['<package_name', 'package_name', ...]}
+    '''
+    if 'pkg.list_provides' in __context__:
+        cached = __context__['pkg.list_provides']
+        return cached
+
+    cmd = ['rpm', '-qa', '--queryformat', '[%{PROVIDES}_|-%{NAME}\n]']
+    ret = {}
+    for line in __salt__['cmd.run'](cmd, output_loglevel='trace', python_shell=False).splitlines():
+        provide, realname = line.split('_|-')
+
+        if provide == realname:
+            continue
+        if provide not in ret:
+            ret[provide] = list()
+        ret[provide].append(realname)
+
+    __context__['pkg.list_list_provides'] = ret
+
+    return ret
+
+def resolve_capabilities(pkgs, refresh, **kwargs):
+    '''
+    .. versionadded:: Oxygen
+
+    Convert name provides in ``pkgs`` into real package names if
+    ``resolve_capabilities`` parameter is set to True. In case of
+    ``resolve_capabilities`` is set to False the package list
+    is returned unchanged.
+
+    refresh
+        force a refresh if set to True.
+        If set to False (default) it depends on zypper if a refresh is
+        executed.
+
+    resolve_capabilities
+        If this option is set to True the input will be checked if
+        a package with this name exists. If not, this function will
+        search for a package which provides this name. If one is found
+        the output is exchanged with the real package name.
+        In case this option is set to False (Default) the input will
+        be returned unchanged.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt '*' pkg.resolve_capabilities resolve_capabilities=True w3m_ssl
+    '''
+    if refresh:
+        refresh_db()
+
+    ret = list()
+    for pkg in pkgs:
+        if isinstance(pkg, dict):
+            for name, version in pkg.items():
+                break
+        else:
+            name = pkg
+            version = None
+
+        if kwargs.get('resolve_capabilities', False):
+            try:
+                search(name, match='exact')
+            except CommandExecutionError as e:
```

Comment:
@cachedout, please update here: we aren't happy with one-character variables, right? I didn't see PyLint is capturing this lately though, but I remember we had requirement somewhere (long time ago though) at least three characters. Is this correct?

Reply:
Hmm, e is unused now. I will remove it.